fix(langchain): correctly initialize huggingface models in init_chat_model

aaron-seq opened the pull request:
**Description:**

This change fixes a `ValidationError` that occurred when using `init_chat_model` with `model_provider="huggingface"`.

The previous implementation incorrectly passed all keyword arguments directly to the `ChatHuggingFace` constructor, which expects an `llm` object rather than a `model_id`. This caused a Pydantic error when arguments like `task` were provided.

The fix updates the logic to first correctly instantiate a `HuggingFacePipeline` object using the `from_model_id` class method, which properly handles arguments like `task`. This initialized pipeline object is then passed to the `ChatHuggingFace` wrapper, resolving the issue.

A unit test has been added to verify the fix and prevent regressions.

Closes #28226

**Issue:**

Fixes #28226

**Dependencies:**

None

aaron-seq committed:
FIX: Correctly initialize HuggingFace models in init_chat_model

aaron-seq committed:
WIP: Saving local changes

aaron-seq committed:
Merge branch 'master' of [https://github.com/langchain-ai/langchain](https://github.com/langchain-ai/langchain)

aaron-seq committed:
Merge branch 'master' into fix/together-tool-message-format

eyurtsev reviewed (commented):
@aaronseq12 thanks for the contribution -- there seem to be a lot of typos in this PR, do you want to try and resolve?

ccurme committed:
Merge branch 'master' into fix/together-tool-message-format

ccurme committed:
remove extraneous changes

aaron-seq committed:
fix: resolve Ruff linting errors for HuggingFace model initialization

- Move exception messages to variables to satisfy EM101 rule
- Use shorter, more concise error messages to satisfy TRY003 rule

aaronseq12 commented:
## 🔧 Linting Issues Fixed

I've resolved the Ruff linting errors that were preventing this PR from passing checks:

### Issues Resolved:

**EM101 Rule Violations:**
- ✅ Moved exception messages to variables instead of using string literals directly in `raise` statements
- ✅ Applied this fix to both `ImportError` and `ValueError` exceptions in the HuggingFace initialization code

**TRY003 Rule Violations:**
- ✅ Shortened exception messages to be more concise and avoid "long messages outside the exception class"
- ✅ Made error messages more specific and actionable

### Changes Made:

1. **ImportError handling:**
   ```python
   # Before: raise ImportError("Please install langchain-huggingface...")
   # After:
   import_error_msg = "Please install langchain-huggingface to use HuggingFace models."
   raise ImportError(import_error_msg) from e
   ```

2. **ValueError handling:**
   ```python
   # Before: raise ValueError("The 'task' keyword argument is required...")
   # After:
   task_error_msg = "The 'task' keyword argument is required for HuggingFace models."
   raise ValueError(task_error_msg)
   ```

The PR should now pass all linting checks while maintaining the same functional behavior for HuggingFace model initialization. 🚀

ccurme committed:
Merge branch 'master' into fix/together-tool-message-format

ccurme committed:
nits
